pkg/minikube/node/config.go

```diff
@@ -156,6 +151,35 @@ func setupKubeconfig(h *host.Host, c *config.ClusterConfig, n *config.Node, clus
 	return kcs, nil
 }
 
+// apiServerURL returns a URL to end user can reach to the api server
+func apiServerURL(h host.Host, cc config.ClusterConfig, n config.Node) (string, error) {
+	hostname := ""
+	port := n.Port
+	var err error
+	if driver.IsKIC(h.DriverName) {
+		// for kic drivers we use 127.0.0.1 instead of node IP,
+		// because of Docker on MacOs limitations for reaching to container's IP.
+		hostname = oci.DefaultBindIPV4
+		port, err = oci.HostPortBinding(h.DriverName, h.Name, port)
+		if err != nil {
+			return "", errors.Wrap(err, "host port binding")
+		}
+		if cc.KubernetesConfig.APIServerName != constants.APIServerName {
```

Comment:
Can we drop this duplicate check? I just know it's going to get refactored out some day.

```
if driver.IsKIC() {
  hostname =
  port = 
} else {
  hostname = h.Driver.GetIP()
}

if cc.KubernetesConfig.APIServerName != constants.APIServerName {
  hostname = cc.KubernetesConfig.APIServerName
}
return fmt.Sprintf("https://" + net.JoinHostPort(hostname, strconv.Itoa(n.Port))), nil
```

Then you can eliminate the duplicate APIServerName check as well as the duplicate URL generation.